src/hooks/useVaaVerifier.ts

```diff
@@ -0,0 +1,128 @@
+import {
+  ChainId,
+  ParsedVaa,
+  hexToUint8Array,
+  parseNFTPayload,
+  parseTransferPayload,
+  parseVaa,
+  tryHexToNativeString,
+} from "@certusone/wormhole-sdk";
+import { useMemo } from "react";
+import {
+  getNFTBridgeAddressForChain,
+  getTokenBridgeAddressForChain,
+} from "../utils/consts";
+
+enum VAAType {
+  UNKNOWN = 0,
+  TRANSFER = 1,
+  ATTEST = 2,
+  TRANSFER_WITH_PAYLOAD = 3,
+}
+
+interface ParserPayload<T> {
+  (buffer: Buffer): T;
+}
+
+function parsePayload<T>(
+  payload: Buffer,
+  parser: ParserPayload<T>
+): T | undefined {
+  try {
+    if (payload) {
+      return parser(payload);
+    }
+  } catch (err) {
+    console.debug(`Failed to parse payload with ${parser}`, err);
+  }
+}
+
+function getPayloadType(payload: Buffer): VAAType {
+  try {
+    const type = payload.readInt8(0);
+    if (type > 0 || type <= 3) {
+      return type;
+    }
+  } catch (err) {
+    console.debug(`Failed to parse type from payload`, err);
+  }
+  return VAAType.UNKNOWN;
+}
+
+function parseVaaFromHexString(hexVaa: string): ParsedVaa {
+  try {
+    return parseVaa(hexToUint8Array(hexVaa));
+  } catch (err) {
+    console.debug(err);
+  }
+  return {} as ParsedVaa;
+}
+
+function itWasEmittedFrom(
+  emitterAddress: Buffer,
+  emitterChain: number,
+  getAddressFn: (chainId: ChainId) => string
+): boolean {
+  try {
+    const chainId = emitterChain as ChainId;
+    const vaaEmitterAddress = tryHexToNativeString(
+      emitterAddress?.toString("hex"),
+      chainId
+    );
+    return (
+      getAddressFn(chainId)?.toUpperCase() === vaaEmitterAddress?.toUpperCase()
+    );
+  } catch (err) {
+    console.debug(err);
+  }
+  return false;
+}
+
+function itIsASupportedType(
```

Comment:
non-important comment: `isSupportedType` name would be better, or `isItASupportedType` if more clarity is wanted

Reply:
oh, good call, I was thinking I've renamed all of them